modules/gmc-agent/src/main/java/org/gridgain/action/controller/BaselineActionsController.java

```diff
@@ -50,12 +57,28 @@ public BaselineActionsController(GridKernalContext ctx) {
     }
 
     /**
-     * @param isAutoAdjustEnabled Is auto adjust enabled.
+     * @param awaitingTime Awaiting time in ms.
+     * @return Completeble feature.
      */
-    public CompletableFuture<Void> updateAutoAdjustAwaitingTime(long isAutoAdjustEnabled) {
+    public CompletableFuture<Void> updateAutoAdjustAwaitingTime(long awaitingTime) {
         authorizeIfNeeded(ctx.security(), SecurityPermission.ADMIN_OPS);
 
-        ctx.grid().cluster().baselineAutoAdjustTimeout(isAutoAdjustEnabled);
+        ctx.grid().cluster().baselineAutoAdjustTimeout(awaitingTime);
+
+        return CompletableFuture.completedFuture(null);
```

Comment:
Looks like that we should execute `ctx.grid().cluster().baselineAutoAdjustTimeout(awaitingTime);` in some thread pool and complete CompletableFuture once method completed.

And other similar places should be reworked.

Reply:
Done.